src/generator.cpp

```diff
@@ -752,6 +773,48 @@ GeneratorInternalVariablePtr Generator::GeneratorImpl::generatorVariable(const V
     return internalVariable;
 }
 
+bool Generator::GeneratorImpl::sameOrEquivalentVariable(const VariablePtr &variable1,
+                                                        const VariablePtr &variable2)
+{
+    // Return whether the given variables are the same or are equivalent (be it
+    // directly or indirectly).
+
+    return (variable1 == variable2) || variable1->hasEquivalentVariable(variable2, true);
+}
+
+GeneratorVariablePtr Generator::GeneratorImpl::variableFirstOccurrence(const VariablePtr &variable,
+                                                                       const ComponentPtr &component)
+{
+    // Recursively look for the first occurrence of the given variable in the
+    // given component.
+
+    GeneratorVariablePtr voi = nullptr;
+
+    for (size_t j = 0; j < component->variableCount(); ++j) {
```

Comment:
I think the break condition should be put into the continuation decision:

`j < component->variableCount() && voi == nullptr`

Reply:
Good point! Done in 4754c2b.

Reply:
This diff is not showing the change in [4754c2b](https://github.com/cellml/libcellml/commit/4754c2bb071fe4b0f66445d6a5daba2afb51e9e5)

Reply:
I guess you hadn't refreshed your page or something... :)